File: server/FONdrum/FONdrum.Domain/Shared/Paginations/PageInfo.cs
namespace FONdrum.Domain.Shared.Paginations
{
    public record PageInfo
    {
        public long TotalCount { get; init; }
        public int? PageSize { get; init; }
        public int? PageNumber { get; init; }

        public long? PagesCount => PageSize.HasValue ? (TotalCount + PageSize.Value - 1) / PageSize.Value : null;
        
        public PageInfo(long totalCount, int? pageSize, int? pageNumber)
        {
            if (totalCount < 0 || pageSize < 0 || pageNumber < 0)
                throw new ArgumentException($"Arguments {nameof(totalCount)}, {nameof(pageSize)}, {nameof(pageNumber)}" +
                    $" must be greater than or equal to 0.");

            TotalCount = totalCount;
            PageSize = pageSize;
            PageNumber = pageNumber;
        }

        public static PageInfo Empty(int? pageSize = null, int? pageNumber = null) => new PageInfo(0, pageSize, pageNumber);
    }
}

File: web-client/fondrum-web-app/src/app/models/shared/pagination/PageInfo.ts
export type PageInfo = {
  totalItemsCount: number;
  totalPagesCount: number;
  pageSize: number;
  pageNumber: number;
};

export const _defaultPageInfo: PageInfo = {
  pageNumber: 1,
  pageSize: 8,
  totalItemsCount: 0,
  totalPagesCount: 0,
};
